tree.go

```diff
@@ -1135,7 +1282,12 @@ func (n *LeafNode) Serialize() ([]byte, error) {
 			}
 		}
 	}
-	return append(append(append([]byte{leafRLPType}, n.stem[:31]...), bitlist[:]...), children...), nil
+	result := make([]byte, 1+31+32, 1+31+32+4*32)
+	result[0] = leafRLPType
+	copy(result[1:], n.stem[:31])
+	copy(result[1+31:], bitlist[:])
+	result = append(result, children...)
```

Comment:
This is a huge saving in memory (which again, less GC pressure and saving GC CPU time that affects overall performance).

The multiple `append` that we did before makes the underlying slice to grow more that we need.
I switched the implementation to by as tight as possible for a size we can use that won't overallocate by the usual `*2` rule of the Go runtime.